MainMenuSample/Source/Game/ButtonFuctions.cs

```diff
@@ -0,0 +1,306 @@
+
+            _audioEnabled = !sfxBool;
+            ToggleAudio(AudioButton.Get<Button>());
+        }
+
+        public override void OnDisable()
+        {
+            PlayButton.Get<Button>().ButtonClicked -= PlayButtonFunction;
+            OptionsMainMenuButton.Get<Button>().ButtonClicked -= ToggleOptionsMenu;
+            OptionsBackButton.Get<Button>().ButtonClicked -= ToggleOptionsMenu;
+            FullscreenButton.Get<Button>().ButtonClicked -= ToggleFullscreen;
+            GraphicsSettingButton.Get<Button>().ButtonClicked -= ModifyGraphicsSettingQuality;
+            AudioButton.Get<Button>().ButtonClicked -= ToggleAudio;
+            QuitButton.Get<Button>().ButtonClicked -= QuitButtonFunction;
+        }
+
+        public string[] ReadSettingsFile()
+        {
+            _lastSettingsReadValue = File.ReadAllLines(_dataPath);
+            return _lastSettingsReadValue;
+        }
+
+        public void WriteSettingsFile(bool GraphicsStatus, bool FullscreenStatus, bool AudioStatus)
+        {
+            string _writtenSettingsValue;
+            string _retreivedSettings;
+
+            if (GraphicsStatus)
+            {
+                _retreivedSettings = GFX_Preset + " = " + SettingsFileTrueTerm + ";";
+            }else
+            {
+                _retreivedSettings = GFX_Preset + " = " + SettingsFileFalseTerm + ";";
+            }
+
+            if (FullscreenStatus)
+            {
+                _retreivedSettings = _retreivedSettings + Environment.NewLine + FullscreenPreset + " = " + SettingsFileTrueTerm + ";";
+            }else
+            {
+                _retreivedSettings = _retreivedSettings + Environment.NewLine + FullscreenPreset + " = " + SettingsFileFalseTerm + ";";
+            }
+
+            if (AudioStatus)
+            {
+                _retreivedSettings = _retreivedSettings + Environment.NewLine + SFX_Preset + " = " + SettingsFileTrueTerm + ";";
+            }else
+            {
+                _retreivedSettings = _retreivedSettings + Environment.NewLine + SFX_Preset + " = " + SettingsFileFalseTerm + ";";
+            }
+
+
+
+            if (EnableSettingsFileComments)
+            {
+                string _settingsComment = "// These values represent the in-game settings." +
+                 Environment.NewLine + "// The values match as following:" + Environment.NewLine + "// Setting enabled -> " + SettingsFileTrueTerm + 
+                 Environment.NewLine + "// Setting disabled -> " + SettingsFileFalseTerm + Environment.NewLine + "// Non-matching values will result in 'False' value.";
+
+                _writtenSettingsValue = _retreivedSettings + Environment.NewLine + Environment.NewLine + _settingsComment; // This is where the 2 bonus newlines come from.
+            }else
+            {
+                _writtenSettingsValue = _retreivedSettings;
+            }
+
+            File.WriteAllText(_dataPath, _writtenSettingsValue);

```

Comment:
The perfect settings setup (see: whatever I posted on Discord) would do this asynchronously instead of potentially blocking things.